### PR TITLE
Boss key hints

### PR DESCRIPTION
**Pull request by Muzugalium**
Added hints that point to the location of the boss key for required dungeons

### CODE REVIEW

File: hints/hint_distribution.py [hunk 1]
```diff
@@ -505,6 +516,21 @@ def _create_random_hint(self):
...
+        loc = self.required_boss_key_locations.pop()
+        if loc in self.hinted_locations:
+            return self._create_bk_hint()
+        self.hinted_locations.append(loc)
+        return LocationGossipStoneHint(
+            loc,
+            self.logic.done_item_locations[loc],
+            True,
+            self.logic.item_locations[loc].get("text"),
+            "boss key",
```
Comment (yannl35133) [inline hints/hint_distribution.py:531]: This string goes into JSON objects, so it's better for spaces to be replaced by `_`

**Review by lepelog — CHANGES_REQUESTED** [review]
- boss key hints have to be added to the changelog
- currently these changes do nothing so bk hints would need to be added to some existing hint distribution or a new one
- if a BK is on LMF BK chest or 3rd small key, that's always hinted and not counted towards the fixed number of BK hints, I guess that's intended?

**Comment by YourAverageLink** [comment]
> - boss key hints have to be added to the changelog
> - currently these changes do nothing so bk hints would need to be added to some existing hint distribution or a new one
> - if a BK is on LMF BK chest or 3rd small key, that's always hinted and not counted towards the fixed number of BK hints, I guess that's intended?

The last two points are likely meant to be solved by new hint distributions. The dungeon always hints would be downgraded to sometimes hints or removed, since the BK hint sort of takes the role those hints used to fill.

**Comment by Muzugalium** [comment]
For the last two points, yep exactly as Yal said

**Review by lepelog — APPROVED** [review]
I still think it's weird to add an unused feature (since it's not used in any distro) but I guess it's fine

**Pull request merged**